Task 14

KoznovDm opened the pull request:
https://t.me/Alexsandrovichhhhh

KoznovDm committed:
add header-verst

KoznovDm committed:
error correction-header DZ

KoznovDm committed:
add font

KoznovDm committed:
add-verst-card
add-verst-footer

KoznovDm committed:
Ts 6 ARRAYS, FUNCTIONS, LOOPS

KoznovDm committed:
Task 8

KoznovDm committed:
Ts8 corrected

KoznovDm committed:
Task 9

KoznovDm committed:
Ts 10

KoznovDm committed:
corrected ts10

KoznovDm committed:
TS_11

KoznovDm committed:
TS_11 corrected

KoznovDm committed:
TS_12

KoznovDm committed:
TS_12

KoznovDm committed:
TS_12 corrected

KoznovDm committed:
TS_12
I changed the validation to an option

KoznovDm committed:
Task_13
design in Angular Material

KoznovDm committed:
Redesigned the interface
Got rid of ngSubmit.

KoznovDm committed:
Task_14
editing a user through a modal window
creating a user through a modal window
deleting a user through a modal window
adding shakbars to all actions

KoznovDm committed:
Task_14
editing a user through a modal window
creating a user through a modal window
deleting a user through a modal window
adding shakbars to all actions

pocahontassss commented on src/app/users-list/users-list.components.ts:
```diff
@@ -34,6 +34,17 @@ export class UserListComponent {
+        ...user,
+        company: {
+          name: user.company.name
+        }
```
Здесь можно передать просто  user, без объекта и ...

pocahontassss commented on src/app/users-list/edit-user-dialog/edit-user-dialog.component.html:
```diff
@@ -0,0 +1,33 @@
+            Название компании введено НЕ ВЕРНО
+        </mat-error>
+    </mat-form-field>
+    <button mat-raised-button color="primary" class="add-user" [mat-dialog-close]="userWithUpdatedFieilds"  [disabled]="formUser.invalid">
```
https://www.youtube.com/watch?v=WTm6VoPu-U8
[1:12:00](https://www.youtube.com/watch?v=WTm6VoPu-U8&t=4320s) - [1:17:20](https://www.youtube.com/watch?v=WTm6VoPu-U8&t=4640s) — Почему нельзя делать привязку через [mat-dialog-close]="submitForm()"

pocahontassss commented on src/app/users.service.ts:
```diff
@@ -23,19 +30,25 @@ export class UsersService {
+        this.snackBar.open('ПОЛЬЗОВАТЕЛЬ УДАЛЕН', 'Закрыть', {
+            duration: 3000,
+        });
+    }
```
Нельзя делать отрисовку UI в сервисах. Сервис отвечает за хранение данных и бизнес-логику, а не за то, что бы отрисовывать какие-то мат бары и так далее. Это должно быть вынесено на уровне компонента

pocahontassss commented on src/app/users-list/users-list.components.ts:
```diff
@@ -45,6 +59,19 @@ export class UserListComponent {
         
   }
 
+    openCreateUserDialog(): void {
+      const dialogRef = this.dialog.open(CreateUserFormComponent, {
+        width: '500px', height: '500px',
+        data: {user: this.user} 
+      });
+
```
А что за юзера мы тут передаем? Это же создания пользователя и на данном этапе его просто не существует

pocahontassss commented on src/app/users-list/users-list.components.ts:
```diff
@@ -21,16 +23,28 @@ import { CreateUserFormComponent } from "../create-user-form/create-user-form.co
     readonly usersService = inject(UsersService);
-
+  
+  @Input()
+  user!: User;
```
Этот инпут здесь не нужен, он в user-card прописывается

pocahontassss commented on src/app/users-list/delete-user-dialog/delete-user-dialog.component.ts:
```diff
@@ -0,0 +1,22 @@
+})
+export class DeleteUserDialogComponent {
+
+  constructor(public dialogRef: MatDialogRef<DeleteUserDialogComponent>) {}
```
Через inject функцию это делается, а не через конструктор. Это старый способ написания, пользуйся новым)

pocahontassss commented on src/app/create-user-form/create-user-form.component.ts:
```diff
@@ -15,6 +17,9 @@ import { MatError } from '@angular/material/form-field';
   styleUrl: './create-user-form.component.scss'
 })
 export class CreateUserFormComponent {
+ readonly data =inject<{ user: User }>(MAT_DIALOG_DATA);
```
Здесь мы никаких данных не получаем, юзера не существует еще

pocahontassss commented on src/app/create-user-form/create-user-form.component.ts:
```diff
@@ -27,12 +32,15 @@ export class CreateUserFormComponent {
+  }
 }
+
+ constructor(public dialogRef: MatDialogRef<CreateUserFormComponent>) {
+  this.formUser.valueChanges.subscribe((formValue) =>
+    console.log('Form Value:', formValue)
+  );
+}}
```
Аналогично. Через inject функцию. И конструктор должен всегда находиться вверху, в начале класса

pocahontassss commented on src/app/create-user-form/create-user-form.component.ts:
```diff
@@ -27,12 +32,15 @@ export class CreateUserFormComponent {
- constructor () {
-  this.formUser.valueChanges.subscribe((formValue) => console.log(formValue))
- }
+public onSubmit(event: Event) {
+  event.preventDefault();
+  if (this.formUser.valid) {
+    this.dialogRef.close(this.formUser.value); 
+  }
```
1.   event.preventDefault(); - это уже js какой-то пошел) лишняя строка
2. Незачем делать проверку на валидность. Если кнопка невалидна - ты не сможешь на нее нажать, никогда.

KoznovDm committed:
Task_14 corrected
in the console, writes when creating and editing "Form submission canceled because the form is not connected"

KoznovDm commented:
привязаю событие отправки формы с помощью (ngSubmit)="submitForm()" к тегу form, тогда корректно обрабатывается отправка формы.

KoznovDm committed:
Task_14 upgrade
